assignments/array-methods.js

```diff
@@ -56,27 +56,47 @@ const runners = [{"id":1,"first_name":"Charmain","last_name":"Seiler","email":"c
 // ==== Challenge 1: Use .forEach() ====
 // The event director needs both the first and last names of each runner for their running bibs.  Combine both the first and last names into a new array called fullName. 
 let fullName = [];
+runners.forEach(function(item){
+    fullName.push(item.first_name, item.last_name);
+});
 console.log(fullName);
 
 // ==== Challenge 2: Use .map() ====
 // The event director needs to have all the runner's first names converted to uppercase because the director BECAME DRUNK WITH POWER. Convert each first name into all caps and log the result
 let allCaps = [];
+runners.map(function(item){
+    allCaps.push(item.first_name.toUpperCase())
+});
 console.log(allCaps); 
 
 // ==== Challenge 3: Use .filter() ====
 // The large shirts won't be available for the event due to an ordering issue.  Get a list of runners with large sized shirts so they can choose a different size. Return an array named largeShirts that contains information about the runners that have a shirt size of L and log the result
 let largeShirts = [];
+largeShirts.push(runners.filter(function(item){
+    return item.shirt_size == "L";
+}));
 console.log(largeShirts);
 
 // ==== Challenge 4: Use .reduce() ====
 // The donations need to be tallied up and reported for tax purposes. Add up all the donations into a ticketPriceTotal array and log the result
 let ticketPriceTotal = [];
+
+let Total = runners.reduce(function(total, item){
+    return total += item.donation;
+}, 0);
+ticketPriceTotal.push(Total);
 console.log(ticketPriceTotal);
 
 // ==== Challenge 5: Be Creative ====
 // Now that you have used .forEach(), .map(), .filter(), and .reduce().  I want you to think of potential problems you could solve given the data set and the 5k fun run theme.  Try to create and then solve 3 unique problems using one or many of the array methods listed above.
 
-// Problem 1
+// Problem 1 - Create an Array Containing runner name and email in object form
+
+let arrObj1 = [];
+arrObj1.push(runners.map((item) => {
+    return {"first_name": item.first_name,"last_name":item.last_name,"email":item.email};
+}));
+console.log(arrObj1);
 
 // Problem 2
```

Comment:
Create a couple more problems and solve them! I'm assuming I just caught you when you weren't quite done here.